osf/models/user.py

```diff
@@ -777,6 +779,24 @@ def merge_user(self, user):
 
         user.save()
 
+    def _merge_users_preprints(self, user):
+        from osf.models.preprint import Preprint, PreprintContributor
+
+        preprints = Preprint.objects.filter(creator=user)
+        for preprint in preprints:
+            preprint.creator = self
+            preprint.save()
```

Comment:
Rather than looping through the all the preprints, it would be faster to use `update`.
```
Preprint.objects.filter(creator=user).update(creator=self)
```